spring-cloud-commons/src/main/java/reactor/core/publisher/FluxFirstNonEmptyEmitting.java

```diff
@@ -0,0 +1,347 @@
+import java.util.stream.Stream;
+
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscription;
+import reactor.core.CoreSubscriber;
+import reactor.core.Scannable;
+import reactor.util.annotation.Nullable;
+
+/**
+ * @author Tim Ysewyn
+ */
+final class FluxFirstNonEmptyEmitting<T> extends Flux<T> implements SourceProducer<T> {
+
+	final Publisher<? extends T>[] array;
+
+	final Iterable<? extends Publisher<? extends T>> iterable;
+
+	@SafeVarargs
+	FluxFirstNonEmptyEmitting(Publisher<? extends T>... array) {
+		this.array = Objects.requireNonNull(array, "array");
+		this.iterable = null;
+	}
+
+	FluxFirstNonEmptyEmitting(Iterable<? extends Publisher<? extends T>> iterable) {
+		this.array = null;
+		this.iterable = Objects.requireNonNull(iterable);
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public void subscribe(CoreSubscriber<? super T> actual) {
+		Publisher<? extends T>[] a = array;
+		int n;
+		if (a == null) {
+			n = 0;
+			a = new Publisher[8];
+
+			Iterator<? extends Publisher<? extends T>> it;
+
+			try {
+				it = Objects.requireNonNull(iterable.iterator(),
+						"The iterator returned is null");
+			}
+			catch (Throwable e) {
+				Operators.error(actual,
+						Operators.onOperatorError(e, actual.currentContext()));
+				return;
+			}
+
+			for (;;) {
+
+				boolean b;
+
+				try {
+					b = it.hasNext();
+				}
+				catch (Throwable e) {
+					Operators.error(actual,
+							Operators.onOperatorError(e, actual.currentContext()));
+					return;
+				}
+
+				if (!b) {
+					break;
+				}
+
+				Publisher<? extends T> p;
+
+				try {
+					p = Objects.requireNonNull(it.next(),
+							"The Publisher returned by the iterator is null");
+				}
+				catch (Throwable e) {
+					Operators.error(actual,
+							Operators.onOperatorError(e, actual.currentContext()));
+					return;
+				}
+
+				if (n == a.length) {
+					Publisher<? extends T>[] c = new Publisher[n + (n >> 2)];
+					System.arraycopy(a, 0, c, 0, n);
+					a = c;
+				}
+				a[n++] = p;
+			}
+
+		}
+		else {
+			n = a.length;
+		}
+
+		if (n == 0) {
+			Operators.complete(actual);
+			return;
+		}
+		if (n == 1) {
+			Publisher<? extends T> p = a[0];
+
+			if (p == null) {
+				Operators.error(actual,
+						new NullPointerException("The single source Publisher is null"));
+			}
+			else {
+				p.subscribe(actual);
+			}
+			return;
+		}
+
+		RaceCoordinator<T> coordinator = new RaceCoordinator<>(n);
+
+		coordinator.subscribe(a, n, actual);
+	}
+
+	/**
+	 * Returns a new instance which has the additional source to be amb'd together with
+	 * the current array of sources.
+	 * <p>
+	 * This operation doesn't change the current FluxFirstEmitting instance.
+	 * @param source the new source to merge with the others
+	 * @return the new FluxFirstEmitting instance or null if the Amb runs with an Iterable
+	 */
+	@Nullable
+	FluxFirstEmitting<T> ambAdditionalSource(Publisher<? extends T> source) {
```

Comment:
careful, you've copied that part without adapting it to the new operator. Note that since you only have a static operator, you don't need that part.